<a href="https://colab.research.google.com/github/rizkiismail9a/data-science-2026-unsia/blob/main/Copy_of_pertemuan_2_MuhamadRizkiIsmail_240401010126.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama: Muhamad Rizki Ismail
NIM: 240401010126
Kelas: IF401

In [1]:
#List
myData = [12, 0, 22, "Budi", "Kucai"]
print(myData[0])
print(myData[-2]) #Dua data dari belakang
print(myData[1: 4]) # Ambil data dari index ke-0 hingga sebelum ke-4

myData.append(10)
myData.insert(1, 55)
myData.remove(12) #Hapus yang value-nya 12
# myData.sort() hanya bisa dilakukan jika tipe data sama
print(myData)

12
Budi
[0, 22, 'Budi']
[55, 0, 22, 'Budi', 'Kucai', 10]


In [2]:
#Dictionary
campaign = {
    'id': 12,
    'title': 'Campaign Saya',
    'campaigner': {
        'name': 'Budi',
        'age': 22
    },
    'location': {
        'id': 123,
        'name': 'Tambakrejo'
    }
}

print(f'Campaigner Name: {campaign["campaigner"]["name"]}')

Campaigner Name: Budi


In [3]:
import numpy as np

# Dari list Python
a1D = np.array([1, 2, 3, 4, 5]) # 1D
a2D = np.array([[1,2,3],[4,5,6]]) # 2D
# Array dengan nilai inisial
zeros = np.zeros((3, 4)) # buat matriks nol, matriks 3x4, 3 baris dan 4 kolom
ones = np.ones((2, 3)) # buat matrisk 1 dengan dua baris dan tiga kolom
eye = np.eye(4) # matriks identitas
full = np.full((2, 2), 7) # nilai konstan

# print(zeros, end='\n\n')
# print(ones, end='\n\n')
# print(eye, end='\n\n')
# print(full, end='\n\n')

# Sequence
arange = np.arange(0, 20, 2) # 0,2,4,...18, dari 0, sampai kurang dari 20, lompat 2
linsp = np.linspace(0, 1, 11) # 11 titik merata
# Array acak (reproducible)
rng = np.random.default_rng(seed=42)
rand_int = rng.integers(0, 100, (3, 4))
rand_flt = rng.normal(loc=0, scale=1, size=(100,))

print(arange, end='\n\n')
print(linsp, end='\n\n')

[ 0  2  4  6  8 10 12 14 16 18]

[0.  0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1. ]



In [4]:
import pandas as pd

# Series dari list dengan index default 0,1,2...
s1 = pd.Series([85, 92, 78, 95, 88, 90, 71, 9, 0, 10, 66])

# Series dengan index costum
nilai = pd.Series([85, 92, 78, 95, 88], index=["Agung", "Ujang", "Rahmat", "Suci", "Rudi"])

# print(nilai['Ujang'], end='\n\n') # → 92
# print(nilai[nilai>80], end='\n\n') # filter
# print(s1.describe(), end='\n\n') # → 87.5
# print(nilai.describe(), end='\n\n') # statistik lengkap


# Membuat DataFrame dari dictionary
data = {
'nama': ['Ali','Budi','Caca','Dina'],
'usia': [22, 25, 21, 23],
'kota': ['Jakarta','Bandung','Surabaya','Medan'],
'ipk': [3.8, 3.5, 3.9, 3.7]
}
myDf = pd.DataFrame(data)

print(myDf.describe)

# Informasi dasar
print(myDf.shape) # → (4, 4)
print(myDf.dtypes) # tipe data per kolom
print(myDf.columns) # nama-nama kolom

<bound method NDFrame.describe of    nama  usia      kota  ipk
0   Ali    22   Jakarta  3.8
1  Budi    25   Bandung  3.5
2  Caca    21  Surabaya  3.9
3  Dina    23     Medan  3.7>
(4, 4)
nama     object
usia      int64
kota     object
ipk     float64
dtype: object
Index(['nama', 'usia', 'kota', 'ipk'], dtype='object')


In [5]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')
penumpang_selamat = df[df['Survived'] == 1]
# jumlah_penumpang_selamat = penumpang_selamat.shape[0] # shape mengembalikan Tuple (pasangan nilai), index pertama jumlah baris dan index kedua menunjukkan jumlah kolom
# print(f'Jumlah penumpang selamat: {jumlah_penumpang_selamat}')
# print(penumpang_selamat)

# Eksplorasi standar setelah load
# print(df.info())
print(df.describe())

       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  


In [6]:
import pandas as pd

titanicDf = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

# method shape mengembalikan Tuple (pasangan nilai), index pertama jumlah baris dan index kedua menunjukkan jumlah kolom
jumlah_baris = titanicDf.shape[0]
jumlah_kolom = titanicDf.shape[1]
print(f'Jumlah baris: {jumlah_baris}')
print(f'Jumlah kolom: {jumlah_kolom}')


Jumlah baris: 891
Jumlah kolom: 12


In [7]:
import pandas as pd

titanicDf = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

# isnull membuat DataFrame baru dengan nilai True di mana ada nilai yang hilang, dan False jika tidak ada.
missing_values = titanicDf.isnull().sum() # return Series
# Mengapa operasi Series dibagi skalar dapat berjalan?
# Ketika Anda membagi Pandas Series dengan skalar (integer len(titanicDf)),
# Pandas akan secara otomatis menerapkan operasi pembagian tersebut ke setiap elemen dalam Series.
# Jadi, setiap jumlah missing value di setiap kolom (yang merupakan nilai dalam Series) akan dibagi dengan total jumlah baris.
missing_percentage = (titanicDf.isnull().sum() / len(titanicDf)) * 100

missing_info = pd.DataFrame({
    'Missing Count': missing_values,
    'Missing Percentage': missing_percentage
})

# Filter to show only columns with missing values
missing_info = missing_info[missing_info['Missing Count'] > 0]

print(missing_info.sort_values(by='Missing Percentage', ascending=False))
print(missing_info.shape)

          Missing Count  Missing Percentage
Cabin               687           77.104377
Age                 177           19.865320
Embarked              2            0.224467
(3, 2)


In [8]:
import pandas as pd

titanicDf = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

survived = titanicDf[titanicDf['Survived'] == 1]
unsurvived = titanicDf[titanicDf['Survived'] == 0]

print(f'Presentase selemat: {(len(survived) / (len(unsurvived) + len(survived))) * 100}')

Presentase selemat: 38.38383838383838


In [9]:
import pandas as pd

titanicDf = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

classOneFemale = titanicDf[(titanicDf['Pclass'] == 1) & (titanicDf['Sex'] == 'female')]
average_age_class_one_female = classOneFemale['Age'].mean()

print(f'Jumlah penumpang wanita kelas 1: {len(classOneFemale)}')
# .2f Meminta agar angka ditampilkan dengan dua digit desimal setelah koma.
print(f'Rata-rata usia penumpang wanita kelas 1: {average_age_class_one_female:.2f} tahun')


Jumlah penumpang wanita kelas 1: 94
Rata-rata usia penumpang wanita kelas 1: 34.61 tahun


In [10]:
import pandas as pd

titanicDf = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

avgSurvivedByClass = titanicDf.groupby('Pclass')['Survived'].mean()
print(avgSurvivedByClass)


aggregationSurvivedByClass = titanicDf.groupby('Pclass').agg(
total = ('PassengerId', 'count'),
selamat = ('Survived', 'sum'),
pct_selamat= ('Survived', 'mean'),
rata_usia = ('Age', 'mean'),
max_fare = ('Fare', 'max')
)

print(aggregationSurvivedByClass.round(2))

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64
        total  selamat  pct_selamat  rata_usia  max_fare
Pclass                                                  
1         216      136         0.63      38.23    512.33
2         184       87         0.47      29.88     73.50
3         491      119         0.24      25.14     69.55


## Kesimpulan

Notebook ini berisi beberapa demonstrasi dasar Python dan pustaka populer seperti NumPy dan Pandas. Bagian utama fokus pada analisis data dataset Titanic.

### Temuan Utama dari Analisis Titanic:

*   **Ukuran Dataset**: Dataset Titanic memiliki 891 baris (penumpang) dan 12 kolom.
*   **Nilai Hilang**: Kolom 'Cabin' memiliki persentase nilai hilang yang sangat tinggi (77.10%), diikuti oleh 'Age' (19.87%) dan 'Embarked' (0.22%).
*   **Tingkat Selamat**: Hanya sekitar 38.38% penumpang yang selamat dari tragedi Titanic.
*   **Penumpang Wanita Kelas 1**: Terdapat 94 penumpang wanita di kelas 1, dengan rata-rata usia sekitar 34.61 tahun.
*   **Tingkat Selamat Berdasarkan Kelas**: Ada perbedaan signifikan dalam tingkat kelangsungan hidup berdasarkan kelas penumpang:
    *   **Kelas 1**: Tingkat selamat tertinggi (62.96%), rata-rata usia 38.23 tahun, dan tarif maksimum 512.33.
    *   **Kelas 2**: Tingkat selamat menengah (47.28%), rata-rata usia 29.88 tahun, dan tarif maksimum 73.50.
    *   **Kelas 3**: Tingkat selamat terendah (24.24%), rata-rata usia 25.14 tahun, dan tarif maksimum 69.55.

Secara keseluruhan, analisis menunjukkan bahwa kelas sosial (Pclass) merupakan faktor yang sangat berpengaruh terhadap peluang selamat di kapal Titanic, dengan penumpang kelas atas memiliki tingkat kelangsungan hidup yang jauh lebih tinggi.